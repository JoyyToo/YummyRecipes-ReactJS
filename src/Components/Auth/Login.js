import React, { Component }  from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField  from 'material-ui/TextField';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';
import {notify} from 'react-notify-toast'
import axiosInstance from '../Constants/Axioscall';
import '../styles.css'


class Login extends Component {
    // initialize state
    constructor (props) {
        super(props);
    this.state = {
        email: '',
        password: '',
        open: false,
    }
}

    // handle user input
    handleInputChange = (event) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({[name]: value});
    }

    // handle login request
    handleLogin = (event) => {
    const payload = new FormData()
    payload.set('email', this.state.email)    
    payload.set('password', this.state.password,)
        
        // send POST request to API
        axiosInstance.post('auth/login', payload)

        .then((response) => {
            localStorage.setItem('token', response.data.jwt_token);
            notify.show(response.data.message, 'success', 4000);
            this.props
            .history
            .push('/');
        })

        .catch((error) => {
            notify.show(error.response.data.message, 'error', 4000);
        });
    }

    // handle reset password request
    handleResetPassword = (event) => {
        const payload = new FormData()
        payload.set('email', this.state.email)    
    
            // send POST request to API
            axiosInstance.post('auth/reset-password', payload)
    
            .then((response) => {
                    notify.show(response.data.message, 'success', 4000);
                    this.props
                    .history
                    .push('/')             
            })
    
            .catch((error) => {         
                notify.show(error.response.data.message, 'error', 4000);
               
            });
        }

    // handle open modal
    handleOpen = () => {
        this.setState({open: true});
      };
    
    // handle close modal
    handleClose = () => {
        this.setState({open: false});
    };

    // render login form
    render(){
        let style = {
            marginLeft: 20,
            width: 323,
          };
        let divstyle = {
            width: 430,
            margin: 'auto',
            textAlign: 'left',
            paddingLeft: 20,
        };
        const actions = [
            <FlatButton
              label="Cancel"
              primary={true}
              onClick={this.handleClose}
            />,
            <FlatButton
              label="Submit"
              primary={true}
              keyboardFocused={true}
              onClick={this.handleResetPassword.bind(this)}
            />,
          ];
        return (
            <MuiThemeProvider>
        <Dialog
          title="Reset Password"
          actions={actions}
          modal={false}
          open={this.state.open}
          onRequestClose={this.handleClose}
        >
          <TextField
            hintText="Email"
            name="email"
            onChange={this.handleInputChange.bind(this)}
            floatingLabelText="Email"
        />

        </Dialog>
            <div className="main">
                <div className="tab-group">
                <Link to="/register">
                    <li className="tab">Sign Up</li></Link>
                <Link to="/login">
                    <li className="tab active">Log In</li></Link>
                </div>
                <div className="inputs">
                <Paper zDepth={2} style={divstyle}>
                    <p className="heading">Login Here</p>
                    <div className="text">
                    <i className="material-icons">mail</i>
                        <TextField floatingLabelText="Email" name="email" 
                                value={this.state.email} style={style} onChange={this.handleInputChange}/><br />

                    <i className="material-icons">lock</i>
                        <TextField floatingLabelText="Password" type="password" name="password" 
                        value={this.state.password} style={style} onChange={this.handleInputChange} /><br /><br /><br />
                    </div>
                    <div className="buttons">
                    <button className="network" id="one" 
                                onClick={(event => this.handleLogin(event))}>LOGIN</button>
                    <Link to="/register"><button className="network" id="two">Cancel</button></Link>
                    
                    <p>Don't have an account? <Link to="/register" type="button" id="button">Register Now!</Link>
                    < br />< br />
                    <button style={{color:'#1ab188', fontSize:'14px'}} 
                                onClick={(event => this.handleOpen(event))}>Forgot password?</ button></p><br />
                    </div>

                </Paper><br />
                    
                </div>
            </div>
            </MuiThemeProvider>
        );
    }}

export default Login;