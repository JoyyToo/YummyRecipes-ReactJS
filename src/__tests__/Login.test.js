import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Login from '../Components/Auth/Login'

describe('Login component', () => {
    const wrapper = shallow(<Login />);
  
    it('renders properly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders 1 <Login/> component', () =>{
        const component = shallow(<Login/>);
        expect(component).toHaveLength(1);
    });

    it('it returns props correctly', () =>{
        const component = shallow(<Login name="app"/>);
        expect(component.instance().props.name).toBe('app')
    });

    it('has initial state', () => {
        expect(wrapper.state().email).toEqual('');
        expect(wrapper.state().password).toEqual('');
    });

    it('has the correct form fields', () => {
        expect(wrapper.find('[name="email"]')).toHaveLength(2);
        expect(wrapper.find('[name="password"]')).toHaveLength(1);
    });

    it('contains divs', () => {
        expect(wrapper.find('div')).toHaveLength(5);
    });

    it('contains li', () => {
        expect(wrapper.find('li')).toHaveLength(2);
    });

    it('contains p', () => {
        expect(wrapper.find('p')).toHaveLength(2);
    });

    it('contains br', () => {
        expect(wrapper.find('br')).toHaveLength(8);
    });

    it('contains in MuiThemeProvider', () => {
        expect(wrapper.find('MuiThemeProvider')).toHaveLength(1);
    });

    it('contains Link', () => {
        expect(wrapper.find('Link')).toHaveLength(4);
    });

    it('contains Paper', () => {
        expect(wrapper.find('Paper')).toHaveLength(1);
    });

    it('contains i', () => {
        expect(wrapper.find('i')).toHaveLength(2);
    });

    it('contains TextField ', () => {
        expect(wrapper.find('TextField')).toHaveLength(3);
    });

    it('contains button', () => {
        expect(wrapper.find('button')).toHaveLength(3);
    });

    it('has initial state', () => {
        expect(wrapper.state().email).toEqual('');
        expect(wrapper.state().password).toEqual('');
    });

    it('has the correct form fields', () => {
        expect(wrapper.find('[name="email"]')).toHaveLength(2);
        expect(wrapper.find('[name="password"]')).toHaveLength(1);
    });
    
})