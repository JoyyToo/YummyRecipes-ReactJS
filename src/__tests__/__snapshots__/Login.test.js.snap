// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders properly 1`] = `
<MuiThemeProvider>
<<<<<<< HEAD
<<<<<<< HEAD
=======
>>>>>>> [chore] Add tests
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
      />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          keyboardFocused={true}
          label="Submit"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Reset Password"
  >
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Email"
      fullWidth={false}
      hintText="Email"
      multiLine={false}
      name="email"
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
    />
  </Dialog>
<<<<<<< HEAD
=======
>>>>>>> [Feature 155119223]:
=======
>>>>>>> [chore] Add tests
  <div
    className="main"
  >
    <div
      className="tab-group"
    >
      <Link
        replace={false}
        to="/register"
      >
        <li
          className="tab"
        >
          Sign Up
        </li>
      </Link>
      <Link
        replace={false}
        to="/login"
      >
        <li
          className="tab active"
        >
          Log In
        </li>
      </Link>
    </div>
    <div
      className="inputs"
    >
      <Paper
        circle={false}
        rounded={true}
        style={
          Object {
            "margin": "auto",
            "paddingLeft": 20,
            "textAlign": "left",
            "width": 430,
          }
        }
        transitionEnabled={true}
        zDepth={2}
      >
        <p
          className="heading"
        >
          Login Here
        </p>
<<<<<<< HEAD
<<<<<<< HEAD
        <div
          className="text"
        >
          <i
            className="material-icons"
          >
            mail
          </i>
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Email"
            fullWidth={false}
            multiLine={false}
            name="email"
            onChange={[Function]}
            rows={1}
            style={
              Object {
                "marginLeft": 20,
                "width": 323,
              }
            }
            type="text"
            underlineShow={true}
            value=""
          />
          <br />
          <i
            className="material-icons"
          >
            lock
          </i>
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Password"
            fullWidth={false}
            multiLine={false}
            name="password"
            onChange={[Function]}
            rows={1}
            style={
              Object {
                "marginLeft": 20,
                "width": 323,
              }
            }
            type="password"
            underlineShow={true}
            value=""
          />
          <br />
          <br />
          <br />
        </div>
=======
        <i
          className="material-icons"
=======
        <div
          className="text"
>>>>>>> [chore] Add tests
        >
          <i
            className="material-icons"
          >
            mail
          </i>
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Email"
            fullWidth={false}
            multiLine={false}
            name="email"
            onChange={[Function]}
            rows={1}
            style={
              Object {
                "marginLeft": 20,
                "width": 323,
              }
            }
            type="text"
            underlineShow={true}
            value=""
          />
          <br />
          <i
            className="material-icons"
          >
            lock
          </i>
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Password"
            fullWidth={false}
            multiLine={false}
            name="password"
            onChange={[Function]}
            rows={1}
            style={
              Object {
                "marginLeft": 20,
                "width": 323,
              }
            }
<<<<<<< HEAD
          }
          type="password"
          underlineShow={true}
          value=""
        />
        <br />
        <br />
        <br />
>>>>>>> [Feature 155119223]:
=======
            type="password"
            underlineShow={true}
            value=""
          />
          <br />
          <br />
          <br />
        </div>
>>>>>>> [chore] Add tests
        <div
          className="buttons"
        >
          <button
            className="network"
            id="one"
            onClick={[Function]}
          >
            LOGIN
          </button>
          <Link
            replace={false}
            to="/register"
          >
            <button
              className="network"
              id="two"
            >
              Cancel
            </button>
          </Link>
          <p>
            Don't have an account? 
            <Link
              id="button"
              replace={false}
              to="/register"
              type="button"
            >
              Register Now!
            </Link>
            <br />
            <br />
<<<<<<< HEAD
<<<<<<< HEAD
            <button
              onClick={[Function]}
=======
            <Link
              onClick={[Function]}
              replace={false}
>>>>>>> [Feature 155119223]:
=======
            <button
              onClick={[Function]}
>>>>>>> [chore] Add tests
              style={
                Object {
                  "color": "#1ab188",
                  "fontSize": "14px",
                }
              }
<<<<<<< HEAD
<<<<<<< HEAD
            >
              Forgot password?
            </button>
=======
              to=""
            >
              Forgot password?
            </Link>
>>>>>>> [Feature 155119223]:
=======
            >
              Forgot password?
            </button>
>>>>>>> [chore] Add tests
          </p>
          <br />
        </div>
      </Paper>
      <br />
    </div>
  </div>
</MuiThemeProvider>
`;
